src/algorithms/post/election_post.md

```diff
@@ -0,0 +1,153 @@
+
+There is no requirement to persist the witnesses (list of merkle proofs) for failure recovery. If something fails, we don’t expect recovery to be fast enough to allow the miner to participate in that round anyway.
+
+## Surprise PoSt cleanup
+
+But while this means a miner will never win blocks from faked power, they won’t be penalized either when storage is lost. How do we ensure that the Power Table is accurate? Likewise, how do we onboard new power since it will not win a block unless it has at least 100TB or 10% of the network.
+
+This is why we need PoSt surprise: 
+
+Atop ElectionPost, a miner will be surprised with a PoSt challenge in every ProvingPeriod (~2 days). The ProvingPeriod resets whenever the miner publishes a PoSt (election or surprise). 
+This SurprisePoSt will use a challenge drawn from the chain at the start of this recovery period. Its PoStProof must be a proof over the PartialTickets for all sectors a miner is storing (i.e. a miner must submit a PoStProof made up of all partialTickets for all eligible sectors, not just the winning ones on sampled sectors). For this reason a miner is incentivized to declare faults in order to successfully generate this PoStProof.
+
+Once challenged, a miner has a given ChallengePeriod (~2 hours) past which they, they will lose their power and a portion of their pledge collateral if they have not submitted a PoSt on chain.
+
+Thereafter, the miner will have three more ProvingPeriods (specifically three challenges, so three ProvingPeriods on expectation) to recover their power by submitting a SurprisePoSt. If the miner does not do so, their sectors are permanently terminated and their storage deal collateral slashed (see the StorageMinerActor in the spec).
+
+Faults are largely independent from this process. The only rules are:
+- Faults cannot be declared during a ChallengePeriod
+- Faults cannot be recovered during a ChallengePeriod
+
+Note that this PoStSurprise will frequently be used by small miners who do not win blocks, and by miners as they are onboarding power to the network, since they will not be able to win any blocks (they have no power) and hence gain power through ElectionPoSt.
+
+Miners earn no reward from submitting PoStSurprise messages. This mechanism does forces a miner to announce their faults and clean up the Power Table accordingly.
+
+Surprise PoSt works as follows:
+```text
+// A number of challenged miners is chosen at every round
+challNumber = NumMiners / ProvingPeriod
+
+// Using the ticket to seed randomness, a miner is picked from the power table for each challenge
+sampledMiners = []
+For i in challNumber:
+    ranHash = H(ticket, i)
+    ranIndex = HashToInt(ranHash) mod len(PowerTable)
+    sampledMiners.append(PowerTable[ranIndex].address)
+```
+Leading to a miner getting randomly sampled once per proving period on expectation.
```

Comment:
I added an explanation. It would be useful for you to read it over and convince yourself it is right. A few design decisions that matter are made here.